torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -395,6 +396,32 @@ def _index_to_param(self) -> Dict[int, torch.Tensor]:
             self._index_to_param_cache = list(chain(*(g["params"] for g in self.param_groups)))
         return self._index_to_param_cache
 
+    @property
+    def _device_to_per_rank_params(self) -> Dict[torch.device, List[List[torch.Tensor]]]:
+        r"""
+        Dict mapping device to a list of the per-rank parameter lists
+        containing the parameters stored on the device.
+
+        ``dev_0`` maps to a list containing:
+            rank 0's assigned parameters stored on ``dev_0``,
+            rank 1's assigned parameters stored on ``dev_0``,
```

Comment:
would I be correct if I assume `dev_0` here means rank 0's and rank1's `dev_0` respectively, and they are different devices?

Reply:
Yes, I meant each `dev_i` to be per rank. I will clarify that in the comment.

Correct me if I am wrong, but I believe that there is an assumption that in this kind of model parallel that the model is always partitioned across devices in the same way for all ranks. This relies on that assumption.